client/solve.go

```diff
@@ -423,13 +423,13 @@ func parseCacheOptions(opt SolveOpt) (*cacheOptions, error) {
 					continue
 				}
 				for _, m := range idx.Manifests {
-					if m.Annotations[ocispec.AnnotationRefName] == "latest" {
+					if m.Annotations[ocispec.AnnotationRefName] == "latest" || attrs["tag"] != "" && m.Annotations[ocispec.AnnotationRefName] == attrs["tag"] {
```

Comment:
`x || y && z` should be written as `x || (y && z)`. 
Also please update README

Reply:
Ideally we should also have test code in https://github.com/moby/buildkit/blob/master/client/client_test.go , though I think it is not mandatory for this case.